### ISSUE TITLE
sudo: sorry, you must have a tty to run sudo (code: 1)

### ISSUE THREAD

**Issue by utgarda** [solutious/rudy#48]
Some error message about sudo ( no sudo calls in my Rudyfile ) :

http://github.com/solutious/rudy/issues/labels/tests
    > rudy --version
    Rudy version: 0.9.8.008
    > rudy startup
     Executing routine: startup 

```
authorize port 22 access for: 194.186.67.200
Created: m-eu-west-1a-stash-stash-01; i-d30a7ea4; 
Waiting for m-eu-west-1a-stash-stash-01 to boot..........^T....
Waiting for public DNS on m-eu-west-1a-stash-stash-01 ...
Waiting for port 22 on m-eu-west-1a-stash-stash-01 .........
root@m-eu-west-1a-stash-stash-01# sudo hostname m-eu-west-1a-stash-stash-01
  sudo: sorry, you must have a tty to run sudo (code: 1)
(S)kip  (R)etry  (F)orce  (A)bort: s
```

**Comment by delano** [comment]
Rudy tries to use sudo for system-level commands when you're not logged in as root. In this case Rudy wanted to set the hostname, but it looks like you are logged in a root. 

For now you can disable this behaviour by adding "hostname :default" to your machines config.  

What is the instance OS?

**Comment by utgarda** [comment]
ami 'ami-55b39821'             # CentOS 5.4, 32bit

**Comment by delano** [comment]
I released a new of Rye (0.9) that won't help with this specific case, but will help with using sudo in your Rudy routines.  Here's a quick rundown:

```
your_routine {
    remote {
        bash {
            sudo '/etc/init.d/nginx', 'reload'
        }
        irb
        tail :f, 'path/2/log'
     }
 }
```

You can run stuff inside a bash shell now. That should open a PTY for sudo. If it still doesn't, the new prompt allows you to drop down into an interactive session ("(S)kip  (R)etry  (I)nteractive  (A)bort:"). When you exit the interactive session, it continues with next command in the routine. 

As a side note you can also connect to a remote IRB session just like it's SSH and run commands like "tail -f" that update the screen while they run.

You grab it with the latest maintenance build of Rudy (0.9.8.009).

**Comment by utgarda** [comment]
"hostname :default" helps, thanks.
